Roommate-Budget-Helper/budget-helper-webui
# Rbh 33 login and signup page

## Jira Ticket: [RBH-33](https://roommatebudgethelper.atlassian.net/browse/RBH-33?atlOrigin=eyJpIjoiNjM5NmM0MDgyYWQ5NGQ4Y2E3NzBjZmYxZTQ3YmU1NmYiLCJwIjoiaiJ9)

## Description of Work
Created login page and connected backend to google and cognito for working sign in.
## Acceptance Criteria

- Registered users can login by username(or emaill) and password or google login
- User logged in is reflected in the session

## How to Test

- Verify google login works for any google user
- Verify a registered user can login and the user is in the session

### List of Changes:

- Added amazon-cognito-identity-js for connecting to cognito
- Fixed issue connecting to database from dev container

### Screenshots

| Image | Description |
| ----- | ----------- |
| <img width="823" alt="unauthenticated" src="https://user-images.githubusercontent.com/7916303/207720596-202023be-b27a-4f0b-93a6-b9b7d7bd18d0.png">| Unauthenticated Login State |
|<img width="828" alt="cognito login" src="https://user-images.githubusercontent.com/7916303/207720589-0e9370df-1847-4b07-a0c0-1ad07e2f44dd.png"> | Successful Cognito Login |
|<img width="828" alt="google login" src="https://user-images.githubusercontent.com/7916303/207720592-da435cfb-1150-4826-8525-9cff54228795.png"> | Successful Google Login |
|<img width="828" alt="invalid login" src="https://user-images.githubusercontent.com/7916303/207720595-468077da-fca8-45bf-9d22-a6b3d3b50f85.png">|Invalid Login|
|<img width="828" alt="unfilled" src="https://user-images.githubusercontent.com/7916303/207721344-54b5a7f4-15c1-415a-bf00-edac92422ff9.png">| Unfilled Login Form |


[RBH-33]: https://roommatebudgethelper.atlassian.net/browse/RBH-33?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
